packages/mui-material-next/migration.md

```diff
@@ -516,3 +516,99 @@ The following example replaces the `MuiCircularProgress-circleDeterminate` class
 - .MuiCircularProgress-circleDeterminate
 + .MuiCircularProgress-determinate .MuiCircularProgress-circle
 ```
+
+## ButtonGroup
+
+### Removed combined styleOverrides keys
+
+The following `styleOverrides` `MuiButtonGroup` keys were removed:
+
+- `contained`
+- `groupedHorizontal`
+- `groupedVertical`
+- `groupedContained`
+- `groupedText`
+- `groupedOutlined`
+- `groupedContainedHorizontal`
+- `groupedTextHorizontal`
+- `groupedOutlinedHorizontal`
+- `groupedContainedVertical`
+- `groupedTextVertical`
+- `groupedOutlinedVertical`
+- `groupedContainedPrimary`
+- `groupedTextPrimary`
+- `groupedOutlinedPrimary`
+- `groupedContainedSecondary`
+- `groupedTextSecondary`
+- `groupedOutlinedSecondary`
+
+The following `styleOverrides` `MuiButtonGroup` keys were added:
+
+- `primary`
+- `secondary`
+- `tertiary`
+- `filled`
+- `filledPartial`
+- `elevated`
+
+You can replace them by using the variants API and CSS Selectors.
+The following example replaces the usage of `groupedOutlined` with the variants API:
+
+```diff
+ const theme = extendTheme({
+   components: {
+     MuiButtonGroup: {
+       styleOverrides: {
+-        groupedOutlined: {
+-          background: "fuchsia"
+-        }
++        outlined: {
++          ".MuiButtonGroup-grouped": {
```

Comment:
Should this one be

```suggestion
+          " .MuiButtonGroup-grouped": {
```

?

Reply:
@DiegoAndai `" .MuiButtonGroup-grouped"` and `".MuiButtonGroup-grouped"` are working here the same. The space is just needed when we use the `&`-selector, which is also possible. Then we have to use `"& .MuiButtonGroup-grouped"` instead of `"&.MuiButtonGroup-grouped"`.